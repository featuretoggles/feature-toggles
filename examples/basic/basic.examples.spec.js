import { transformFileSync } from "@babel/core";
import { transformFileSync as transformFS } from "babel-core";
import featureTogglePlugin from "../../src";

/**
 * Here this is a basic example get transpile by babel plugin
 */
it("BABEL7 - Basic function", () => {
  const { code } = transformFileSync(`${__dirname}/src/basic.js`, {
    rootMode: "upward",
    plugins: [[featureTogglePlugin, { dir: __dirname, toggleName: "ft1" }]]
  });
  expect(code).toMatchSnapshot();
});
it("BABEL6 - Basic function", () => {
  const { code } = transformFS(`${__dirname}/src/basic.js`, {
    plugins: [[featureTogglePlugin, { dir: __dirname, toggleName: "ft1" }]] //Option will not work here
  });
  expect(code).toMatchSnapshot();
});

it("BABEL7- React component function ", () => {
  const { code } = transformFileSync(`${__dirname}/src/ReactComponent.js`, {
    rootMode: "upward",
    plugins: [
      [featureTogglePlugin, { dir: __dirname, toggleName: "ft1" }],
      "@babel/plugin-transform-react-jsx"
    ]
  });
  expect(code).toMatchSnapshot();
});
it("BABEL7 - React component function with Argv", () => {
  const { code } = transformFileSync(`${__dirname}/src/ReactComponent.js`, {
    rootMode: "upward",
    plugins: [
      [featureTogglePlugin, { dir: __dirname }],
      "@babel/plugin-transform-react-jsx"
    ]
  });
  expect(code).toMatchSnapshot();
});
it("BABEL6 - React component function with Argv", () => {
  const { code } = transformFS(`${__dirname}/src/ReactComponent.js`, {
    plugins: [[featureTogglePlugin, { dir: __dirname }], "transform-react-jsx"]
  });
  expect(code).toMatchSnapshot();
});
it("BABEL7 - Vue component function with Argv", () => {
  const { code } = transformFileSync(`${__dirname}/src/VueComponent.js`, {
    rootMode: "upward",
    plugins: [
      [featureTogglePlugin, { dir: __dirname }],
      "@babel/plugin-transform-react-jsx"
    ]
  });
  expect(code).toMatchSnapshot();
});
it("BABEL6 - Vue component function with Argv", () => {
  const { code } = transformFS(`${__dirname}/src/VueComponent.js`, {
    plugins: [[featureTogglePlugin, { dir: __dirname }], "transform-react-jsx"]
  });
  expect(code).toMatchSnapshot();
});
