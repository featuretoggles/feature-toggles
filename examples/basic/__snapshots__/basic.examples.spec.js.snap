// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BABEL6 - Basic function 1`] = `
"// \`
// If \\"ft1\\" toggle is used to build the application:
// {
//     \\"feature-2\\" : true,
//     \\"feature-3\\" : false // Any code wrapped with feature-3 will be removed
// }
// \`

/**
 * If you wan to reassign a variable
 * please use below toggles
 */
let abc;
abc = true;
/* toggleStart(feature-3) */

/* toggleEnd(feature-3) */

/**
 * SWITCH CASE
 */
switch (abc) {
  case true:
    console.log(\\"Hello\\");
  /* toggleStart(feature-3) */
}

/**
 * Inside Object
 */
const obj = {
  /* toggleEnd(feature-3) */
  xyz: true
};

/**
 * Object Member Function
 */
obj
/* toggleStart(feature-3) */
.func()
/* toggleEnd(feature-3) */
.func1();

/**
 * If else condition
 */

if (abc &&
//toggleEnd(feature-3)
obj.abc) {
  console.log();
}
//toggleStart(feature-3)
else {}
//toggleEnd(feature-3)

function basic() {
  // toggleEnd(feature-3)

  // The below toggle is not available in the toggle list so that will not removed by toggle
  // toggleStart(feature-10)
  console.log(\\"This will not remove\\");
  // toggleEnd(feature-10)
}

export default {
  basic
};"
`;

exports[`BABEL6 - React component function with Argv 1`] = `
"/*
featureTogglesConfig: {
  \\"feature-10\\": false
}
*/
import React from \\"react\\";

/* toggleStart(feature-10) */

/* toggleEnd(feature-10) */

class Welcome extends React.Component {
  render() {
    return React.createElement(
      \\"div\\",
      null,
      React.createElement(
        \\"h1\\",
        null,
        \\"You only see me, I am a full feature\\"
      ),
      React.createElement(
        \\"h1\\",
        null
      )
    );
  }
}
export const newComponent = () => {
  return React.createElement(
    \\"div\\",
    null,
    React.createElement(
      \\"h1\\",
      null,
      \\"This feature is existing feature\\"
    ),
    React.createElement(
      \\"h1\\",
      null,
      \\"You only see me, I am a full feature\\"
    ),
    React.createElement(Welcome, null),
    React.createElement(
      \\"h1\\",
      null,
      \\"This will show\\"
    )
  );
};
export default Welcome;"
`;

exports[`BABEL6 - Vue component function with Argv 1`] = `
"let template;
/**
 * When we want to add a new feature we should wrap the old code with a \\"old-\\" prefix toggleName.
 * This will be easy to remove code automatically using codemod.
 */
/* toggleStart(old-feature-3) */

/* toggleEnd(feature-3) */
Vue.component({});"
`;

exports[`BABEL7 - Basic function 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = void 0;
// \`
// If \\"ft1\\" toggle is used to build the application:
// {
//     \\"feature-2\\" : true,
//     \\"feature-3\\" : false // Any code wrapped with feature-3 will be removed
// }
// \`

/**
 * If you wan to reassign a variable
 * please use below toggles
 */
var abc;
abc = true;
/* toggleStart(feature-3) */

/* toggleEnd(feature-3) */

/**
 * SWITCH CASE
 */
switch (abc) {
  case true:
    console.log(\\"Hello\\");

  /* toggleStart(feature-3) */
}
/**
 * Inside Object
 */


var obj = {
  /* toggleEnd(feature-3) */
  xyz: true
};
/**
 * Object Member Function
 */

obj
/* toggleStart(feature-3) */
.func()
/* toggleEnd(feature-3) */
.func1();
/**
 * If else condition
 */

if (abc && //toggleEnd(feature-3)
obj.abc) {
  console.log();
} //toggleStart(feature-3)
else {} //toggleEnd(feature-3)


function basic() {
  // toggleEnd(feature-3)
  // The below toggle is not available in the toggle list so that will not removed by toggle
  // toggleStart(feature-10)
  console.log(\\"This will not remove\\"); // toggleEnd(feature-10)
}

var _default = {
  basic: basic
};
exports[\\"default\\"] = _default;"
`;

exports[`BABEL7 - React component function with Argv 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = exports.newComponent = void 0;

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));

var _createClass2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));

var _possibleConstructorReturn2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));

var _getPrototypeOf2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));

var _inherits2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));

var _react = _interopRequireDefault(require(\\"react\\"));

/*
featureTogglesConfig: {
  \\"feature-10\\": false
}
*/

/* toggleEnd(feature-10) */
var Welcome = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2[\\"default\\"])(Welcome, _React$Component);

  function Welcome() {
    (0, _classCallCheck2[\\"default\\"])(this, Welcome);
    return (0, _possibleConstructorReturn2[\\"default\\"])(this, (0, _getPrototypeOf2[\\"default\\"])(Welcome).apply(this, arguments));
  }

  (0, _createClass2[\\"default\\"])(Welcome, [{
    key: \\"render\\",
    value: function render() {
      return _react[\\"default\\"].createElement(\\"div\\", null, _react[\\"default\\"].createElement(\\"h1\\", null, \\"You only see me, I am a full feature\\"), _react[\\"default\\"].createElement(\\"h1\\", null, \\"This will show\\"));
    }
  }]);
  return Welcome;
}(_react[\\"default\\"].Component);

var newComponent = function newComponent() {
  return _react[\\"default\\"].createElement(\\"div\\", null, _react[\\"default\\"].createElement(\\"h1\\", null, \\"This feature is existing feature\\"), _react[\\"default\\"].createElement(\\"h1\\", null, \\"You only see me, I am a full feature\\"), _react[\\"default\\"].createElement(Welcome, {
    abc: \\"true\\"
    /* toggleStart(feature-3) */

  }), _react[\\"default\\"].createElement(\\"h1\\", null, \\"This will show\\"));
};

exports.newComponent = newComponent;
var _default = Welcome;
exports[\\"default\\"] = _default;"
`;

exports[`BABEL7 - Vue component function with Argv 1`] = `
"\\"use strict\\";

var template;
/**
 * When we want to add a new feature we should wrap the old code with a \\"old-\\" prefix toggleName.
 * This will be easy to remove code automatically using codemod.
 */

/* toggleStart(old-feature-3) */

/* toggleEnd(feature-3) */
Vue.component(\\"button-counter\\", {
  data: function data() {
    return {
      count: 0
    };
  },
  template: template
});"
`;

exports[`BABEL7- React component function  1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = exports.newComponent = void 0;

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));

var _createClass2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));

var _possibleConstructorReturn2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));

var _getPrototypeOf2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));

var _inherits2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));

var _react = _interopRequireDefault(require(\\"react\\"));

/*
featureTogglesConfig: {
  \\"feature-10\\": false
}
*/

/* toggleEnd(feature-10) */
var Welcome = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2[\\"default\\"])(Welcome, _React$Component);

  function Welcome() {
    (0, _classCallCheck2[\\"default\\"])(this, Welcome);
    return (0, _possibleConstructorReturn2[\\"default\\"])(this, (0, _getPrototypeOf2[\\"default\\"])(Welcome).apply(this, arguments));
  }

  (0, _createClass2[\\"default\\"])(Welcome, [{
    key: \\"render\\",
    value: function render() {
      return _react[\\"default\\"].createElement(\\"div\\", null, _react[\\"default\\"].createElement(\\"h1\\", null, \\"You only see me, I am a full feature\\"), _react[\\"default\\"].createElement(\\"h1\\", null, \\"This will show\\"));
    }
  }]);
  return Welcome;
}(_react[\\"default\\"].Component);

var newComponent = function newComponent() {
  return _react[\\"default\\"].createElement(\\"div\\", null, _react[\\"default\\"].createElement(\\"h1\\", null, \\"This feature is existing feature\\"), _react[\\"default\\"].createElement(\\"h1\\", null, \\"You only see me, I am a full feature\\"), _react[\\"default\\"].createElement(Welcome, {
    abc: \\"true\\"
    /* toggleStart(feature-3) */

  }), _react[\\"default\\"].createElement(\\"h1\\", null, \\"This will show\\"));
};

exports.newComponent = newComponent;
var _default = Welcome;
exports[\\"default\\"] = _default;"
`;
